src/math.jl

```diff
@@ -14,10 +14,10 @@ for (op,fn) in ((:+,:af_add), (:.+,:af_add), (:-,:af_sub), (:.-,:af_sub), (:.*,:
                 (:.<<, :af_bitshiftl),(:>>, :af_bitshiftr), (:.>>, :af_bitshiftr),
                 (:^, :af_pow), (:.^, :af_pow ))
 
-    @compat @eval function $op{T,S}(a::AFArray{T}, b::AFArray{S}; batched = true)
+    @compat @eval function $op{T,S, N1, N2}(a::AFArray{T,N1}, b::AFArray{S,N2}; batched = true)
         ptr = new_ptr()
         eval($(quot(fn)))(ptr, a, b, batched)
```

Comment:
Do you not need that `eval` to call the appropriate function? Is there a cleaner way to writing this, then? 😄